src/trees/FunctionTree.cpp

```diff
@@ -586,31 +875,31 @@ template <int D> int FunctionTree<D>::crop(double prec, double splitFac, bool ab
  * Also returns an array with the corresponding indices defined as the
  * values of serialIx in refTree, and an array with the indices of the parent.
  * Set index -1 for nodes that are not present in refTree */
-template <int D>
-void FunctionTree<D>::makeCoeffVector(std::vector<double *> &coefs,
+template <int D, typename T>
+void FunctionTree<D, T>::makeCoeffVector(std::vector<T *> &coefs,
                                       std::vector<int> &indices,
                                       std::vector<int> &parent_indices,
                                       std::vector<double> &scalefac,
                                       int &max_index,
-                                      MWTree<D> &refTree,
-                                      std::vector<MWNode<D> *> *refNodes) {
+                                      MWTree<D, double> &refTree,
```

Comment:
here (and some other places) you explicitly fix the scalar type of some function inputs. Why is that?